### PR TITLE
[WIP] [Python Bindings] Changed Vector/Matrix map to Python objects

### PR DESCRIPTION
**Pull request by francesco-romano**
tl;dr; instead of using Python function to generate and parse string to convert between types, we now use native C code.

This might have an impact on current Python code. For sure, it will require some modifications:

- removed the helper functions as they are no longer needed
- renamed `fromList` to `fromPython` (as it now accepts both lists and Numpy arrays).

Tests are missing. I do not have the bandwidth to implement them. 
Let me know what do you think - if something is missing.

Of course, waiting for travis to catch all the bugs 😄

### CODE REVIEW

File: bindings/python/python.i [hunk 1]
```diff
@@ -1,75 +1,265 @@
...
+
+%include "numpy.i"
+
+%init %{
+import_array();
+%}
+
 %{
 #include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/VectorFixSize.h>
```
Comment (francesco-romano) [inline bindings/python/python.i]: @traversaro not sure I need to include all these files. For example, I forgot to include `Wrench.h`, etc. and it seems to work.

**Comment by traversaro** [comment]
cc @diegoferigo @anqingd and all the Python users around.

**Comment by traversaro** [comment]
There are a few users of the iDynTree's Python API . I would like to implement proper deprecation of the old APIs before removing them.

**Comment by francesco-romano** [comment]
Should be enough to add a `fromList` that maps to `fromPython` I think. It can be done in the old `init_helpers`

**Comment by francesco-romano** [comment]
While working on some tests, I realised the PR is not ready as `fromPython` should be a sort of factory method, while now it is a simple method acting on an already-built object.

**Comment by francesco-romano** [comment]
Can somebody run the python tests?
I tried with:
- Python 2.7.10 (default on my Mac 10.13.6) and I get a SegFault
- Python 2.7.11 (on travis) same error as before
- Python 2.7.13 and it works

**Comment by traversaro** [comment]
Handy recap of Python version on different systems: https://repology.org/metapackage/python2/versions .

**Comment by diegoferigo** [comment]
I tried a rebased version of this branch on Ubuntu 18.04 with Python 3.6 and I've got the following failures:

```
The following tests FAILED:
         51 - PythonHelperTests (SEGFAULT)
         52 - PythonDeprecatedTests (SEGFAULT)
```

After a quick execution in the interpreter of the additional binded methods I discovered that `fromNumpy()` and `toNumpy()` work well. The segfault seems due the methods that use regular python `list`s.

cc @francesco-romano @traversaro

**Comment by francesco-romano** [comment]
Happy if you are able to rebase this and merge it. It has remained here for far too long.

This problem does not look new to me. I am sure I read something similar recently somewhere. I think there has been some changes to Python lists in some 3.x version. Hopefully I will find again the discussion

**Comment by traversaro** [comment]
@diegoferigo Did you worked on this recently as part of the development of Python-based controllers?

**Comment by diegoferigo** [comment]
@traversaro I am using a rebased version of this branch since some while without any problem. Though, segfaults without any errors occur if they're not used correctly, and debugging these problems can be challenging in a Python project. I think this PR is a great start, but it needs some polishing to become stable enough.

I keep postponing this activity due to other priorities, but I have to do it before releasing the first stable release of gym-ignition in any case.

**Comment by traversaro** [comment]
As this PR is currently stale, I will close it, but feel free to re-open it if someone wants to work on this, thanks!

**Pull request closed without merging**